velox/type/Filter.cpp

```diff
@@ -280,6 +280,35 @@ bool BigintRange::testingEquals(const Filter& other) const {
       (upper_ == otherBigintRange->upper_);
 }
 
+bool BigintRange::testBloomFilter(
+    const AbstractBloomFilter& bloomFilter,
+    const velox::Type& type) const {
+  // Don't test bloom filter for a wider range
+  if ((upper_ - lower_) > kMaxBloomFilterChecks)
+    return true;
+
+  switch (type.kind()) {
+    case TypeKind::INTEGER:
+      for (int32_t val = lower32_; val <= upper32_; ++val) {
+        if (bloomFilter.mightContainInt32(val)) {
```

Comment:
mightContainInt32 is virtual, and you're calling it on every value. This would incur unnecessary virtual function dispatch cost. We can create a batch interface like `bool mightContainInt32(values, numValues)`, so that there will only be one virtual function dispatch for testing all values. Same for other filters.